Convert datetimes to UTC

roback opened the pull request:
- `Query#start_time` and `#end_time` are converted to UTC before making the request.
- `Post#published` and `#indexed` has their timezone set to UTC.

close #16 
### Todo
- [x] test that use dateutil (to show that that works/is supported) (98bcf5e911e06a364f176b50271a9d62f609c510)
- [x] show in examples that you can use pytz or dateutil (cdecac5fc8942d3bb435e8af8d1b250f8106a809)

roback committed:
Convert Query#start_time and #end_time to UTC

close #16

roback committed:
Remove duplicated code

The methods Query#_ts and Query#_tsTo were almost the same.
Also, _tsTo used camelcase which is another reason I wanted to get
rid if it :)

roback committed:
Include timezone in Post#published and #indexed

Query#start_time and #end_time requires the user of the gem to set
timezone, we should set it ourselves too.

roback committed:
Always use #_parse_time helper function in Post

Self can not be accessed outside functions so the defaults has
to be set in a constructor. This change also fixes the following bug:
The attributes declared outside of the constructor, as they were before,
resulted in the attributes beeing set on the class and not the instance.

roback committed:
Set timezone in examples

walro commented on tests/test_query.py:
```diff
@@ -41,19 +46,45 @@ def test_query_should_add_language(self):
+        q = self._client.query()
+        q.pattern = "spotify"
+        q.start_time = datetime(2012, 12, 28, 9, 1, 22)
+        with self.assertRaises(twingly_search.TwinglyQueryException):
```
Maybe you've talked about this already, but if no timezone given I think we should assume UTC rather than raising.

dentarg replied:
I said we should raise: https://github.com/twingly/twingly-search-api-python/issues/16#issuecomment-187186010

but in the next comment I started to doubt myself... so it's still open for discussion

walro replied:
Ah, thought I had seen it somewhere. Anyway, what does our other libraries do? I was under the impression we don't raise, but I haven't closely followed the latest changes and I can't test now.

To me, at least, it makes sense to assume no tzinfo => UTC.

dentarg replied:
It's different for Ruby and Node.js, because they default to your local timezone, so we always have that information. We haven't done PHP yet, but PHP will yell at you if you don't configure your timezone or explicitly pick one when creating the date object (as we do in our examples).

roback replied:
Since the standard library has no support for timezones I think we could just assume its UTC. Right now the user is required to add pytz/dateutil (or something similar) to be able to use the client.

dentarg replied:
Sounds reasonable.

We can have something about it in the README.

roback replied:
Made some changes in 868a5c8c97597f345b65a2bc67e604ed4af4fc4c

roback committed:
Assume UTC if no timezone is set

roback committed:
Fix typo in readme

dentarg commented on requirements.txt:
```diff
@@ -1,2 +1,3 @@
 future
 requests
+pytz
```
we also list requirements in the README, but don't I think we need to do that when we have this file

roback replied:
Yes, I think that requests should be removed from the readme instead :)

dentarg replied:
Done in https://github.com/twingly/twingly-search-api-python/commit/13bef577aea8f11528f1d6e8b7b929b835c0fae4

roback replied:
:+1:

dentarg commented on test_requirements.txt:
```diff
@@ -1,4 +1,5 @@
 future
 requests
+pytz
```
we don't need to duplicate requirements, see https://github.com/html5lib/html5lib-python/blob/master/requirements-test.txt :smile:

roback replied:
Cool, I'll change :)

dentarg commented:
What we talked about during fika: 
- test that use dateutil (to show that that works/is supported)
- show in examples that you can use pytz or dateutil

roback committed:
No need to duplicate requirements

roback committed:
Add tests with time parsed by dateutil

There are two popular date-time related packages in python;
pytz and dateutil. Now we have query tests for both of them.

roback committed:
Add example that uses local timezone

An example explaining how twingly_search handles timezones.

roback commented:
> show in examples that you can use pytz or dateutil

The examples I added only uses dateutil (I went with dateutil because I wanted to use its `parse()` function).

dentarg committed:
Merge branch 'master' into utc

dentarg committed:
No need to repeat what's in requirements.txt

dentarg commented:
LGTM :)
